src/EditorFeatures/Text/Extensions.SnapshotSourceText.cs

```diff
@@ -334,25 +338,19 @@ public override IReadOnlyList<TextChangeRange> GetChangeRanges(SourceText oldTex
                     }
                 }
 
-                var oldSnapshot = oldText.FindCorrespondingEditorTextSnapshot();
-                var newSnapshot = this.FindCorrespondingEditorTextSnapshot();
+                var oldSnapshot = oldText.FindCorrespondingEditorTextImage();
+                var newSnapshot = this.FindCorrespondingEditorTextImage();
                 return GetChangeRanges(oldSnapshot, oldText.Length, newSnapshot);
             }
 
-            private IReadOnlyList<TextChangeRange> GetChangeRanges(ITextSnapshot oldSnapshot, int oldTextLength, ITextSnapshot newSnapshot)
+            private IReadOnlyList<TextChangeRange> GetChangeRanges(ITextImage oldSnapshot, int oldTextLength, ITextImage newSnapshot)
             {
                 if (oldSnapshot == null ||
-                    newSnapshot == null ||
-                    oldSnapshot.TextBuffer != newSnapshot.TextBuffer)
+                    newSnapshot == null)
                 {
                     // Claim its all changed
                     Logger.Log(FunctionId.Workspace_SourceText_GetChangeRanges, "Invalid Snapshots");
-                    return ImmutableArray.Create<TextChangeRange>(new TextChangeRange(new TextSpan(0, oldTextLength), this.Length));
-                }
-                else if (oldSnapshot.Version.ReiteratedVersionNumber == newSnapshot.Version.ReiteratedVersionNumber)
```

Comment:
ITextImageVersion doesn't have ReiteratedVersionNumber.  As above ,we could try to map back to ITextSnapshots and preserve this optimization if we succeed.  What do people think?